### DISCUSSION TITLE
Keep user_data_dir after browser close

### DISCUSSION THREAD

**Discussion by awaixom** [ultrafunkamsterdam/undetected-chromedriver#1910] [General]
Code:

`
class Bot:
    
    main_tab: uc.Tab
    
    def __init__(self, email, password, proxy): # proxy = host:port or user:pass:ip:port
      self.email = email
      self.password =  password
      self.proxy = proxy
      self.browser = None
      self.current_page = None
      self.profile_fodler_path = os.path.join(os.getcwd(),'zzprofile')
      if not os.path.exists(self.profile_fodler_path):
        os.mkdir(self.profile_fodler_path)
      uc.loop().run_until_complete(self.run())

    async def auth_challenge_handler(self, event: uc.cdp.fetch.AuthRequired):
        # Split the credentials
        # Respond to the authentication challenge
        USERNAME = self.proxy.split(":")[0]
        PASSWORD = self.proxy.split(":")[1]
        asyncio.create_task(
            self.main_tab.send(
                uc.cdp.fetch.continue_with_auth(
                    request_id=event.request_id,
                    auth_challenge_response=uc.cdp.fetch.AuthChallengeResponse(
                        response="ProvideCredentials",
                        username=USERNAME,
                        password=PASSWORD,
                    ),
                )
            )
        )

    async def req_paused(self, event: uc.cdp.fetch.RequestPaused):
        asyncio.create_task(
            self.main_tab.send(
                uc.cdp.fetch.continue_request(request_id=event.request_id)
            )
        )
    
    def print_me(self,msg):
      msg = f'[{self.email}] {msg}'
      print(msg)
      
    async def load_driver(self):
      PROXY = f"{self.proxy.split(':')[2]}:{self.proxy.split(':')[3]}"
      self.browser = await uc.start(
          browser_args=[f"--proxy-server={PROXY}"],
          headless=False,
          user_data_dir=self.profile_fodler_path,
      )
      self.main_tab = await self.browser.get("draft:,")
      self.main_tab.add_handler(uc.cdp.fetch.RequestPaused, self.req_paused)
      self.main_tab.add_handler(
          uc.cdp.fetch.AuthRequired, self.auth_challenge_handler
      )
      await self.main_tab.send(uc.cdp.fetch.enable(handle_auth_requests=True))

      `

everytime when my program stop, it automatically remove/delete the user_data_dir from my computer. I want to keep it and i dont want it to delete it.
Please help !

**Comment by michaelwj** [comment]
I had a similar issue, but got it to work as a browser arg:

`browser_args = [f"--user-data-dir=USER_DATA_DIR"]`

I'm persisting some browser extension configs this way.
